library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationSource.java

```diff
@@ -4,133 +4,91 @@
 
-        void onResult(@NonNull LocationSettingsResult result);
-
-        void onLocationChanged(Location location);
+        void onLocationResult(@Nullable LocationResult locationResult);
     }
 
     GooglePlayServicesLocationSource(Context context, LocationRequest locationRequest, SourceListener sourceListener) {
         this.sourceListener = sourceListener;
         this.locationRequest = locationRequest;
-        this.googleApiClient = new Builder(context)
-              .addApi(LocationServices.API)
-              .addConnectionCallbacks(this)
-              .addOnConnectionFailedListener(this)
-              .build();
-    }
-
-    boolean isGoogleApiClientConnected() {
-        return googleApiClient.isConnected();
-    }
-
-    void connectGoogleApiClient() {
-        googleApiClient.connect();
-    }
-
-    void disconnectGoogleApiClient() {
-        googleApiClient.disconnect();
-    }
-
-    void clearGoogleApiClient() {
-        googleApiClient.unregisterConnectionCallbacks(this);
-        googleApiClient.unregisterConnectionFailedListener(this);
-
-        if (googleApiClient.isConnected()) {
-            removeLocationUpdates();
-        }
-
-        googleApiClient.disconnect();
+        this.fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(context);
     }
 
     void checkLocationSettings() {
-        LocationServices.SettingsApi
-              .checkLocationSettings(googleApiClient, new LocationSettingsRequest.Builder()
-                    .addLocationRequest(locationRequest)
-                    .build())
-              .setResultCallback(this);
-    }
-
-    void startSettingsApiResolutionForResult(Status status, Activity activity) throws SendIntentException {
-        status.startResolutionForResult(activity, RequestCode.SETTINGS_API);
+        LocationServices.getSettingsClient(fusedLocationProviderClient.getApplicationContext())
+                .checkLocationSettings(
+                        new LocationSettingsRequest.Builder()
+                                .addLocationRequest(locationRequest)
+                                .build()
+                )
+                .addOnSuccessListener(new OnSuccessListener<LocationSettingsResponse>() {
+                    @Override
+                    public void onSuccess(LocationSettingsResponse locationSettingsResponse) {
+                        if (sourceListener != null) sourceListener.onSuccess(locationSettingsResponse);
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception exception) {
+                        if (sourceListener != null) sourceListener.onFailure(exception);
+                    }
+                });
+    }
+
+    void startSettingsApiResolutionForResult(@NonNull ResolvableApiException resolvable, Activity activity) throws SendIntentException {
```

Comment:
Let's follow the Android structure here, please have the `Activity` as first parameter in the method

Reply:
Sorry just noticed that's how it was originally, so you may ignore this comment if you like to.
Have no idea why I have done that way in the first place 🤦